src/cli/cli-application.ts

```diff
@@ -0,0 +1,40 @@
+import { Command } from './commands/command.interface.js';
+import { CommandParser } from './command-parser.js';
+
+type CommandCollection = Record<string, Command>;
+
+export class CLIApplication {
+  private commands: CommandCollection = {};
+
+  constructor(private readonly defaultCommand: string = '--help') {}
+
+  public registerCommands(commandList: Command[]): void {
+    commandList.forEach((command) => {
+      if (Object.hasOwn(this.commands, command.getName())) {
```

Comment:
Object.hasOwn(this.commands, command.getName()) => this.commands[command.getName]